synapse/visibility.py

```diff
@@ -130,9 +142,10 @@ def allowed(event: EventBase) -> Optional[EventBase]:
             sender_erased=erased_senders.get(event.sender, False),
         )
 
```

Comment:
Why does the `state_groups` table exist?

```sh
$ psql synapse
#  \d+ state_groups
                                Table "public.state_groups"
  Column  |  Type  | Collation | Nullable | Default | Storage  | Stats target | Description
----------+--------+-----------+----------+---------+----------+--------------+-------------
 id       | bigint |           | not null |         | plain    |              |
 room_id  | text   |           | not null |         | extended |              |
 event_id | text   |           | not null |         | extended |              |
```

It seems like the subset of `state_groups_state`


```sh
$ psql synapse
#  \d+ state_groups_state
                               Table "public.state_groups_state"
   Column    |  Type  | Collation | Nullable | Default | Storage  | Stats target | Description
-------------+--------+-----------+----------+---------+----------+--------------+-------------
 state_group | bigint |           | not null |         | plain    |              |
 room_id     | text   |           | not null |         | extended |              |
 type        | text   |           | not null |         | extended |              |
 state_key   | text   |           | not null |         | extended |              |
 event_id    | text   |           | not null |         | extended |              |
```